src/gfx/pal_spec.cpp

```diff
@@ -226,13 +228,51 @@ static void readLine(std::filebuf &file, std::string &buffer) {
 /*
  * Parses the initial part of a string_view, advancing the "read index" as it does
  */
-static uint16_t parseDec(std::string const &str, std::string::size_type &n) {
-	uint32_t value = 0; // Use a larger type to handle overflow more easily
+template<typename T>
+static std::optional<T> parseDec(std::string const &str, std::string::size_type &n) {
+	std::string::size_type start = n;
+
+	uintmax_t value = 0; // Use a larger type to handle overflow more easily
 	for (auto end = std::min(str.length(), str.find_first_not_of("0123456789", n)); n < end; ++n) {
-		value = std::min<uint32_t>(value * 10 + (str[n] - '0'), UINT16_MAX);
+		value = std::min(value * 10 + (str[n] - '0'), (uintmax_t)std::numeric_limits<T>::max);
 	}
 
-	return value;
+	return n > start ? std::optional<T>{value} : std::nullopt;
+}
+
+static std::optional<Rgba> parseColor(std::string const &str, std::string::size_type &n, uint16_t i) {
+	std::optional<uint8_t> r = parseDec<uint8_t>(str, n);
+	if (!r) {
+		error("Failed to parse color #%" PRIu16 " (\"%s\"): invalid red component", i + 1,
+		      str.c_str());
+		return std::nullopt;
+	}
+	skipWhitespace(str, n);
+	if (n == str.length()) {
+		error("Failed to parse color #%" PRIu16 " (\"%s\"): missing green component", i + 1,
+		      str.c_str());
+		return std::nullopt;
+	}
```

Comment:
I feel like the "missing" errors are redundant with the "invalid" ones. Remove them?